drivers/ipm/ipm_imx.c

```diff
@@ -81,7 +148,7 @@ static void imx_mu_isr(const struct device *dev)
 	 * Store immediate overlapping exception return operation
 	 * might vector to incorrect interrupt
 	 */
-#if defined __CORTEX_M && (__CORTEX_M == 4U)
+#if (defined __CORTEX_M) && ((__CORTEX_M == 4U) || (__CORTEX_M == 7U))
```

Comment:
I was wondering why __DSB is also required for Cortex-M7.
While searching for possible reasons, I came across this comment:
```
    /* ARM errata 838869, affects Cortex-M4, Cortex-M4F Store immediate overlapping
     * exception return operation might vector to incorrect interrupt.
     * For Cortex-M7, if core speed much faster than peripheral register write speed,
     * the peripheral interrupt flags may be still set after exiting ISR, this results to
     * the same error similar with errata 83869 */
```
If that was the reason to enable this for M7 as well, it would be nice to mention that in the commit.

Thanks for your work!

Reply:
@emfend Good find! When adding this support for the rev2 MU peripheral, I used the RPMsg-Lite examples from the MCUXpresso SDK for this target (the i.MX8M Plus SoC) as a reference, and that comment is from the MU IRQ handler in [`rpmsg_platform.c`](https://github.com/NXPmicro/rpmsg-lite/blob/master/lib/rpmsg_lite/porting/platform/imx8mp_m7/rpmsg_platform.c#L89-L111):

```c++
/*
 * MU Interrrupt RPMsg handler
 */
int32_t MU1_M7_IRQHandler(void)
{
    uint32_t channel;

    if ((((1UL << 27U) >> RPMSG_MU_CHANNEL) & MU_GetStatusFlags(MUB)) != 0UL)
    {
        channel = MU_ReceiveMsgNonBlocking(MUB, RPMSG_MU_CHANNEL); // Read message from RX register.
        env_isr(channel >> 16);
    }
    /* ARM errata 838869, affects Cortex-M4, Cortex-M4F Store immediate overlapping
     * exception return operation might vector to incorrect interrupt.
     * For Cortex-M7, if core speed much faster than peripheral register write speed,
     * the peripheral interrupt flags may be still set after exiting ISR, this results to
     * the same error similar with errata 83869 */
#if (defined __CORTEX_M) && ((__CORTEX_M == 4U) || (__CORTEX_M == 7U))
    __DSB();
#endif

    return 0;
}
```

I'll add the second half the comment from `rpmsg_platform.c` to the one in this file like you suggest. Thanks!

Reply:
I came across this while writing a driver [1] for exactly the same MU :)

Since I was afraid that it would become confusing if another driver was packed into the existing file, I decided to pack this into a separate file. But from what I've seen, it's quite ok.
Since you were faster, your version will go upstream anyway.
Thanks for your work!

[1] https://github.com/emfend/zephyr/tree/imx8mp-mu